File: Easy/bit_positions.py
data = ['86,2,3', '125,1,2']

def dec_to_bits(n):
	bits = ''
	while n > 0:
		m = n%2
		n = n // 2
		bits = bits + str(m)
	return bits[::-1]


#import sys
#test_cases = open(sys.argv[1], 'r')
#data = test_cases.readlines()
for test in data:
	l = test.split(",")
	dec_num = int(l[0])
	bits_num = dec_to_bits(dec_num)
	
	i = int(l[1]) - 1
	j = int(l[2]) - 1
	K = list(bits_num[::-1])
	if K[i] == K[j]:
		print 'true'
	else:
		print 'false'

#test_cases.close()

File: Easy/bit_positions.r
test.cases <- c("86,2,3", "125,1,2")

dec_to_bits <- function(n){
	bits <- c()
	while (n > 0){
		m <- n %% 2
		n <- floor(n / 2)
		bits <- c(bits, as.character(m))
	}
	rev(bits)
}

#args <- commandArgs(trailingOnly=TRUE)
#test.cases <- strsplit(readLines(args[[1]], warn=FALSE), '\n')
cat(sapply(test.cases, function(test) {
	if (length(test) > 0) {
		test <- strsplit(test, split = ",")
		dec <- as.integer(test[[1]][1])
		pos1 <- as.integer(test[[1]][2])
		pos2 <- as.integer(test[[1]][3])
		binary <- rev(dec_to_bits(dec))
		if (binary[pos1] == binary[pos2]){
			"true"
		}else{
			"false"
		}
	}
}), sep = "\n")